### PR TITLE
Task 1

### PR DESCRIPTION
**Pull request by sheraz176**

### CODE REVIEW

File: app/Http/Controllers/taskController.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Task;
+use App\Models\tasks;
+class taskController extends Controller
+{
+    
+ function addtask(Request $req){
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/taskController.php]: Function name must be camelcase `addTask`

File: app/Http/Controllers/taskController.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      $tasks->name=$req->name;
+    
+     
+      $tasks->save();
+
+      return redirect('/');
+     }
+
+ function fatchtask(){
+        $item = tasks::orderBy('id', 'desc')->take(5)->paginate(10);
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/taskController.php:23]: Function name must be camelcase `fatchTask`

File: app/Http/Controllers/taskController.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        return view('/tasks',['tasks'=>$item]);
+  }
+
+  function delete($id){
+        $data=tasks::find($id);
```
Comment (AsfandYarAhmad) [inline app/Http/Controllers/taskController.php]: It should be `findOrFail`

File: app/Models/Task.php [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Task extends Model
+{
+    use HasFactory;
+
+ public $table="tasks";
+      public $timestamps=false;
+
```
Comment (AsfandYarAhmad) [inline app/Models/Task.php]: Why using models two times?

File: database/migrations/2022_10_01_175721_create_tasks_table.php [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('tasks', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('name');
```
Comment (AsfandYarAhmad) [inline database/migrations/2022_10_01_175721_create_tasks_table.php]: Length not set for `name` and cannot be nullable

File: resources/views/tasks.blade.php [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  <body>
+
+  <div class="container">
+<div class="row">
+<div class="col-md-12">
+<form action="addtask" method="POST"  enctype="multipart/form-data"  class="card-body" tabindex="500">
+									@csrf
+  <div class="form-group">
+    <label for="exampleInputEmail1">Task</label>
+    <input type="text" class="form-control"  name="name" placeholder="Enter Task">
```
Comment (AsfandYarAhmad) [inline resources/views/tasks.blade.php]: Name should be required

File: resources/views/tasks.blade.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+  
+
+    <!-- Optional JavaScript -->
+    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.12.9/dist/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.0.0/dist/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+  </body>
+</html>
```
Comment (AsfandYarAhmad) [inline resources/views/tasks.blade.php:54]: indentation is very bad

File: routes/web.php [hunk 8]
```diff
@@ -1,18 +1,30 @@
...
-/*
-|--------------------------------------------------------------------------
-| Web Routes
-|--------------------------------------------------------------------------
-|
-| Here is where you can register web routes for your application. These
-| routes are loaded by the RouteServiceProvider within a group which
-| contains the "web" middleware group. Now create something great!
-|
-*/
```
Comment (AsfandYarAhmad) [inline routes/web.php:14]: why you remove this

File: routes/web.php [hunk 9]
```diff
@@ -1,18 +1,30 @@
...
+ 
+/**
+ * Add A New Task
+ */
+
+/**
+ * Delete An Existing Task
+ */
+Route::delete('/task/{id}', function ($id) {
+    //
```
Comment (AsfandYarAhmad) [inline routes/web.php]: Why you add this here?

File: resources/views/common/errors.blade.php [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+ 
+        <br><br>
+ 
+        <ul>
+            @foreach ($errors->all() as $error)
+                <li>{{ $error }}</li>
+            @endforeach
+        </ul>
+    </div>
+@endif
```
Comment (AsfandYarAhmad) [inline resources/views/common/errors.blade.php]: This is not used why create this?

File: resources/views/layouts/app.blade.php [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    <body>
+        <div class="container">
+            <nav class="navbar navbar-default">
+                <!-- Navbar Contents -->
+            </nav>
+        </div>
+ 
+        @yield('content')
+    </body>
+</html>
```
Comment (AsfandYarAhmad) [inline resources/views/layouts/app.blade.php:16]: This is not used why create this?

File: resources/views/tasks.blade.php [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+
+  
+
+    <!-- Optional JavaScript -->
+    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/popper.js@1.12.9/dist/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.0.0/dist/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+  </body>
```
Comment (AsfandYarAhmad) [inline resources/views/tasks.blade.php]: Why add these scripts?

Comment (chvlog) [reply resources/views/tasks.blade.php:72]: Javascript Bootstrap 4 use to design these secript

File: resources/views/layouts/app.blade.php (hunk 11 above)
Comment (chvlog) [reply resources/views/layouts/app.blade.php:19]: yes this is not include beacuse i include files not responsive design

File: resources/views/common/errors.blade.php (hunk 10 above)
Comment (chvlog) [reply resources/views/common/errors.blade.php:15]: yes error. php files not included. beacuse validation in controler files

File: database/migrations/2022_10_01_175721_create_tasks_table.php (hunk 5 above)
Comment (chvlog) [reply database/migrations/2022_10_01_175721_create_tasks_table.php:17]: ok

File: app/Models/Task.php (hunk 4 above)
Comment (chvlog) [reply app/Models/Task.php:14]: yes beacuse i make a controller. i add functions in controler. model use in database table connectivity

File: app/Http/Controllers/taskController.php (hunk 2 above)
Comment (chvlog) [reply app/Http/Controllers/taskController.php:24]: ok

File: app/Http/Controllers/taskController.php (hunk 1 above)
Comment (sheraz176) [reply app/Http/Controllers/taskController.php]: ok i create all functions in controler

File: database/migrations/2022_10_01_175721_create_tasks_table.php (hunk 5 above)
Comment (sheraz176) [reply database/migrations/2022_10_01_175721_create_tasks_table.php]: function name null able beacuse database issues in null value

File: routes/web.php (hunk 8 above)
Comment (sheraz176) [reply routes/web.php:14]: ok my mistake

File: resources/views/tasks.blade.php (hunk 12 above)
Comment (sheraz176) [reply resources/views/tasks.blade.php]: beacuse use bootstrap 4 responsive design

File: resources/views/layouts/app.blade.php (hunk 11 above)
Comment (sheraz176) [reply resources/views/layouts/app.blade.php:16]: i include files not responsive design

File: resources/views/common/errors.blade.php (hunk 10 above)
Comment (sheraz176) [reply resources/views/common/errors.blade.php]: i use validation in controler files

File: routes/web.php (hunk 9 above)
Comment (sheraz176) [reply routes/web.php]: Delete function in Route folders

File: app/Models/Task.php (hunk 4 above)
Comment (sheraz176) [reply app/Models/Task.php]: use beacuse i make a controller. i write function in controler. model use as a same name database connectivity

File: app/Http/Controllers/taskController.php (hunk 3 above)
Comment (sheraz176) [reply app/Http/Controllers/taskController.php]: ?

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 13]
```diff
@@ -15,7 +15,7 @@ public function up()
     {
         Schema::create('users', function (Blueprint $table) {
             $table->id();
-            $table->string('name');
+            $table->string('name')->nullable();
             $table->string('email')->unique();
             $table->timestamp('email_verified_at')->nullable();
             $table->string('password');
```
Comment (AsfandYarAhmad) [inline database/migrations/2014_10_12_000000_create_users_table.php]: What is this?

**Review by AsfandYarAhmad — APPROVED** [review]

**Pull request merged**